backend/index.js

```diff
@@ -108,70 +143,281 @@ function createLaptopBody(req){
+        res.status(400).send(ERRORS.INVALID_PARAMETERS);
+      }
+      if (!PASSWORD_REGEX.test(password)) { 
+        res.status(400).send(ERRORS.PASSWORD_COMPLEXITY);
+      }
+
+      crypto.pbkdf2(password, salt, 310000, 32, "sha256", function(err, hashedPassword) {
+        if (err) { return cb(err); }
+        let query = "UPDATE users SET password = ?, salt = ? WHERE id = ?";
+        db.query(query, [hashedPassword, salt, id], (err, result) => {
+          if (err) next(err);
+          res.send();
+        });
+      });
+    } else { 
+      res.status(401).send();
+    }
+  } else { 
+    if (req.user[0].role === ROLES.ADMIN || req.user[0].role === ROLES.SUPERADMIN) {
+      
+      // Do not allow role changes to self
+      if (requestUserId === id) { 
+        res.status(400).send(ERRORS.USER_CHANGES_FAIL);
+      }
+      if (typeof role !== "string") { 
+        res.status(400).send(ERRORS.INVALID_PARAMETERS);
+      }
+      if (!ALPHANUMERIC_REGEX.test(role)) { 
+        res.status(400).send(ERRORS.INVALID_PARAMETERS);
+      }
+
+      let query = "UPDATE users SET role = ? WHERE id = ?";
+      db.query(query, [role, id], (err, result) => {
+        if (err) next(err);
+        res.send();
+      });
+    } else {
+      res.status(401).send();
+    }
+  }
+});
+
+// Delete user
+app.delete("/users/:id", authJwt, isAdmin, (req, res, next) => {
+  const id = parseInt(req.params.id);
+
+  if (typeof id !== "number") {
+    res.status(400).send(ERRORS.INVALID_PARAMETERS);
+  }
+
+  let getUserQuery = "SELECT id, role FROM users WHERE id = ?";
+  db.query(getUserQuery, [id], (err, result) => {
+    if (result.length === 0) { 
+      res.status(404).send(ERRORS.INVALID_PARAMETERS);
+    }  
+    else if (result[0].role === ROLES.SUPERADMIN) { 
+        res.status(400).send(ERRORS.INVALID_PARAMETERS);
+      } else {
+        let deleteUserQuery = "DELETE FROM users WHERE id = ?";
+
+        db.query(deleteUserQuery, [id], (err, result) => {
+          if (err) next(err);
+          res.send();
+        })
+      }
+  }); 
+});
+
+// Get all laptops
+app.get("/inventory", authJwt, (req, res) => {
+  let sqlQuery = "SELECT * FROM laptops";
+
+  db.query(sqlQuery, (err, results) => {
+    if (err) next(err);
+    res.send(results);
+  });
+});
+
+// Get laptop by Id
+app.get("/inventory/:id", authJwt, (req, res) => {
+  const id = parseInt(req.params.id);
+
+  if (typeof id !== "number") { 
+    res.status(400).send(ERRORS.INVALID_PARAMETERS);
+  }
+
+  let sqlQuery = "SELECT * FROM laptops WHERE id = ? LIMIT 1";
+
+  db.query(sqlQuery, [id], (err, results) => {
+    if (err) return next(err)
+
+    if (results.length === 0) {
+      return res.status(404).send("Laptop with serial number", id, "not found.");
+    }
+    return res.send(results[0]);    
+  });
+});
+
+// Add a new laptop 
+// TODO: Consider adding type checks once laptop class is better defined
+app.post("/add", authJwt, (req, res, next) => {
```

Comment:
Are these parameters use or is it handled in the middleware now?

Reply:
Any references to authJwt or authLocal indicate that the route requires authentication (either through JWT or local username + password). This isn't used as a parameter necessarily but acts as a flag for the middleware to intercept.